src/link_assembly_files.py

```diff
@@ -23,20 +23,27 @@
 import testing
 
 
-def create_outname(outdir, infile):
+def create_outname(outdir, infile, extras):
   """Create the output file's name."""
   basename = os.path.basename(infile)
-  outname = basename + '.wast'
+  linker = os.path.splitext(os.path.basename(extras['linker']))[0]
+  if linker == 'lld':
+    outname = basename + '.wasm'
+  else:
+    outname = basename + '.wast'
   return os.path.join(outdir, outname)
 
 
 def link(infile, outfile, extras):
   """Create the command-line for a linker invocation."""
   linker = extras['linker']
   basename = os.path.splitext(os.path.basename(linker))[0]
+  install_root = os.path.dirname(os.path.dirname(linker))
+  symfile = os.path.join(install_root, 'lib', 'syms.def')
   commands = {
+      'lld': [linker, '-flavor', 'wasm', '-entry=main',
+        '--allow-undefined-file='+ symfile, '-o', outfile, infile],
```

Comment:
Nit: Should probably align the whitespace of the second line to the `[`, like

```python
'lld': [linker, '-flavor', 'wasm', '-entry=main',
        '--allow-undefined-file='+ symfile, '-o', outfile, infile],
```

Nit: the spacing around the `+` isn't consistent.